H-1652, H-1693: Add ability to bulk action drafts

benwerner01 opened the pull request:
## 🌟 What is the purpose of this PR?

<!-- Explain, at a high level, what this does and why. -->
<!-- Use the 'What does this change?' section to list more specific implementation details. -->

This PR add the ability to bulk accept/ignore draft entities on the `/drafts` page.

As a driveby, it also increases the width of the accept draft link warning modal (H-1693).

## 🔗 Related links

<!-- Add links to any context it is worth capturing (e.g. Issues, Discussions, Discord) -->
<!-- Mark any links which are not publicly accessible as _(internal)_ -->
<!-- Don't rely on links to explain the PR, especially internal ones: use the sections above -->

- H-1652
- H-1693

## 🔍 What does this change?

<!-- Use a bullet list to explain your changes in more detail, if it would be helpful. -->
<!-- If applicable, link to the specific commit.-->

- updates the checkbox styling of the MUI `Checkbox` component to match those from the hash.ai figma

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?

<!-- Confirm you have taken the necessary action to record a changeset or publish a change, as appropriate -->
<!-- Tick AT LEAST ONE box and delete the rest. Do not delete this section! see libs/README.md for info on publishing -->

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

<!-- If this adds a user facing feature or modifies how an existing feature is used, it likely needs a docs change. -->
<!-- Tick ONE box and delete the rest. Do not delete this section! -->

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

<!-- If this adds or moves an existing package, modifies `scripts` in a `package.json`, it likely needs a turbo graph change. -->
<!-- Tick ONE box and delete the rest. Do not delete this section! -->

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

<!-- Are there known issues / intentionally omitted functionality? Flag them here to save reviewers doing so -->

## 🐾 Next steps

<!-- Are there are planned/suggested follow ups which are related but won't be done in this PR? -->

## 🛡 What tests cover this?

<!-- What automated tests cover this? Existing ones? New ones? None? -->

Manual testing.

## ❓ How to test this?

<!-- Tell reviewers how they can test the functionality -->

Try to bulk ignore or accept draft entities on the drafts page. To test the warning modals, you will need to have at least one draft link linked to a draft entity.

## 📹 Demo

<!-- Add a screenshot or video showcasing your work -->

https://github.com/hashintel/hash/assets/42802102/4353f705-2ab8-4fac-98fc-3fed37d580ef

Warning modal with increased width:

<img width="881" alt="image" src="https://github.com/hashintel/hash/assets/42802102/771ca19f-70e5-4954-a9de-f0880d5a093f">

nonparibus commented:
@benwerner01 Please can we make the cells hug their contents? In this example "Has Parent" is overly wide.

> Warning modal with increased width:
> 
> <img alt="image" width="881" src="https://private-user-images.githubusercontent.com/42802102/294480989-771ca19f-70e5-4954-a9de-f0880d5a093f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0NTg1ODYsIm5iZiI6MTcwNDQ1ODI4NiwicGF0aCI6Ii80MjgwMjEwMi8yOTQ0ODA5ODktNzcxY2ExOWYtNzBlNS00OTU0LWE5ZGUtZjA4ODBkNWEwOTNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA1VDEyMzgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxYzRmMWJlYjU5OTU3NDE0ZDMwOTY4ZjA4NTZlOGQwOGM2ZDM3ODk1ZWNlYWExMDVjYTM1MGY2NDE1ZTE1NjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z-lJrrEbIj_reBCiOb6Pxh9OCKodCqI83LaGQ1dmcOo">

benwerner01 commented:
> @benwerner01 Please can we make the cells hug their contents? In this example "Has Parent" is overly wide. e.g. [see here](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH?type=design&node-id=10522%3A73005&mode=design&t=mN6vIKgRwYUs1VyB-1) (internal Figma link)

Yes, I've reverted the behaviour:

<img width="733" alt="image" src="https://github.com/hashintel/hash/assets/42802102/a2c26fd5-09c6-4eda-ab2f-60675e331a03">

I've also fixed the chips from overflowing when the entity label is too long by cutting the text shorter via an ellipsis, took longer than expected because I forgot about `min-width`/`min-height` is set to `auto` for flex children by default 🤦‍♂️

<img width="736" alt="image" src="https://github.com/hashintel/hash/assets/42802102/2725d414-8c7d-4362-81b8-13afea78ace4">



> Can we also take care to all-caps text-transform the "Draft" and "Live" labels on the entity cells?

Addressed this in https://github.com/hashintel/hash/pull/3829/commits/3ede3074de832fe528c6591c91afa21b939a9629

CiaranMn reviewed (approved):
Greart work @benwerner01!
